osu.Game/Rulesets/Scoring/ScoreProcessor.cs

```diff
@@ -333,21 +333,21 @@ public double ComputeScore(ScoringMode mode, ScoringValues current, ScoringValue
         /// <param name="totalBasicHitObjects">The total number of basic (non-tick and non-bonus) hitobjects in the beatmap.</param>
         /// <returns>The total score computed from the given scoring component ratios.</returns>
         [Pure]
-        public double ComputeScore(ScoringMode mode, double accuracyRatio, double comboRatio, double bonusScore, int totalBasicHitObjects)
+        public long ComputeScore(ScoringMode mode, double accuracyRatio, double comboRatio, long bonusScore, int totalBasicHitObjects)
         {
             switch (mode)
             {
                 default:
                 case ScoringMode.Standardised:
                     double accuracyScore = accuracyPortion * accuracyRatio;
                     double comboScore = comboPortion * comboRatio;
-                    return (max_score * (accuracyScore + comboScore) + bonusScore) * scoreMultiplier;
```

Comment:
Turns out this and `ScoringMode.Classic`'s return value need rounding (for example at `TestFourVariousResultsOneMiss` it returns `387499.99999999994` instead of `387500`). Before the type change they only got rounded in `PopulateScore()`.

Reply:
I'm still not sure about your change here. I meant that you'd want this to _not_ be rounded for `Classic`, so the implementation would be something like:

```
long GetScore()
{
    if (Standardised)
        return (long)Math.Round(getScoreRaw(...));
    else
        return (long)Math.Round(getScoreRaw(...) * classic_stuff);

    double getScoreRaw(...)
    {
        // Raw standardised score.
    }
}
```

Reply:
Oh, sorry. I'll be pushing a fix in a few minutes.